### PR TITLE
new test class created

### LINKED ISSUE #16: Testing
Write tests for the features.

### PR DESCRIPTION
**Pull request by kimmari1**
- Created new instrumented test class to test database user operations

### CODE REVIEW

**Review by peterjangml — COMMENTED** [review]
Great progress! I believe the test script is ready to be merged. However, please see below for detailed comments:

`DatabaseInstrumentedTest`
- Nice instrumented test for the user database!
- I am wondering why this is an instrumented test instead of a unit test. Is it because we need to access MainActivity in order to test the database?

`DatabaseHelper`
- Great addition of getUsername method
- Can I ask if it's possible to create removeUser? It would be helpful to test all the prompts
    - If not, it is okay

`RegisterUser` and `SigninUser`
- I believe these changes are conflicting with PR #31 
- Can we resolve this together?

**Comment by kimmari1** [comment]
> 

Yes, we can create removeUser! I'm not sure how this would look like in the app, as we don't have any way for the user to delete accounts but it can still be added in. 

For RegisterUser and SigninUser, the conflict comes from me adding in lines when another commit worked on those same lines. I thought I resolved the conflict by shifting down the code from PR#31 by a couple lines, but if there are still conflicts, I can try to take a look again. The only change I made was setting the variable getUsername to the getter method created in DatabaseHelper.

File: app/src/main/java/com/example/tester/RegisterUser.java [hunk 1]
```diff
@@ -31,8 +31,25 @@ protected void onCreate(Bundle savedInstanceState) {
 
             try {
                 registerUserValidation.validateRegister(myDB, username, password, repassword);
+
                 Toast.makeText(RegisterUser.this, "Registration Successful.", Toast.LENGTH_SHORT).show();
```
Comment (peterjangml) [inline app/src/main/java/com/example/tester/RegisterUser.java]: Here, it doesn't seem to be aligned with our current main branch.
Android Toasts are all currently replaced with Prompts, but it seems to be alive in this PR.

File: app/src/main/java/com/example/tester/SigninUser.java [hunk 2]
```diff
@@ -29,10 +29,29 @@ protected void onCreate(Bundle savedInstanceState) {
 
             try {
                 signinValidation.validateSignin(userRepo, username, password);
+
                     Toast.makeText(SigninUser.this, "Sign-in Successful.", Toast.LENGTH_SHORT).show();
```
Comment (peterjangml) [inline app/src/main/java/com/example/tester/SigninUser.java]: Same situation in `SigninUser` as `RegisterUser`. Here, it doesn't seem to be aligned with our current main branch.
Android Toasts are all currently replaced with Prompts, but it seems to be alive in this PR.

**Comment by kimmari1** [comment]
My bad, just resolved the conflicts and committed my changes. The code should be good to merge now.

**Review by peterjangml — APPROVED** [review]
No worries! Thanks for the revision :)

**Pull request merged**